docs/chymyst07.md

```diff
@@ -783,10 +793,173 @@ go { case barrier1(x1, reply1) + barrier2(x2, reply2) => ... }
+This will unblock the previous reaction,
+ 
+```scala
+{ case barrier(_, replyB) + counter(6, replyC) => replyC(); counter(7); replyB() }
+
+```
+
+which was blocked at the call to `counter(7)`.
+Now this reaction can proceed to reply to `barrier()` using `replyB()`.
+However, at this point it is too early to send a reply to `barrier()`, because we still have two `barrier()` molecules that we have not yet consumed!
+
+Therefore, `reaction1` is incorrect.
+On the other hand, `reaction2` works correctly.
+It will block at each emission of `counter(k+1)` and unblock only when `k=n-1`.
+At that time, `reaction2` will reply to the `counter(n-1)` molecule and to the last consumed `barrier()` molecule.
+The reply to the `counter(n-1)` molecule will unblock the copy of `reaction2` that emitted it, which will unblock the next molecules.
+
+In this way, the `n`-rendezvous will require all `n` reactions to wait at the "barrier" until all participants reach the barrier, and then unblock all of them.
+
+It remains to see how the first reaction will start.
+We could emit a `counter(0)` molecule at the initial time.
+This molecule is blocking, so emitting it will block a thread until the very end of the `n`-rendezvous.
+We could avoid blocking a thread by writing a special initial reaction such as
+
+```scala
+go { case barrier(_, replyB) + counterInit(_) => counter(1); replyB() }
+
+```
+
+The complete code looks like this:
+
+```scala
+val barrier = b[Unit,Unit]
+val counterInit = m[Unit]
+val counter = b[Int,Unit]
+
+site(
+  go { case barrier(_, replyB) + counterInit(_) => // this reaction will consume the very first barrier molecule emitted
+    counter(1) // one reaction has reached the rendezvous point
+    replyB()
+  },
+  go { case barrier(_, replyB) + counter(k, replyC) =>
+    if (k + 1 < n) counter(k+1)
+    replyC()
+    replyB()
+  }
+)
+counterInit() // This needs to be emitted initially.
+
+```
+
+Note that this chemistry will block `n` threads that emit `barrier()` and another set of `n` threads that run `reaction2`.
+Unless (in a future version of `JoinRun`) the blocking is optimized away, the current implementation of the `n`-rendezvous will require `2*n` threads and block them until the rendezvous is passed.
+
+### Refactoring into a function
+
+As usual when encapsulating some piece of chemistry into a reusable library function, we first figure out what new molecule emitters are necessary for the users of the library to be able to use the encapsulated chemistry.
+These new emitters will be returned by the library function; all the other chemistry will remain encapsulated within the local scope of the library function.
+
+In our case, the users of the library only need the `barrier()` molecule emitter.
+In fact, users should _not_ have access to `counter` or `counterInit` emitters: if users emit some more of these molecules, the logic of the reactions will be disrupted.
```

Comment:
will be disrupted could be replaced with "will be undefined" or "will be violated". Disrupted is still fine if you like it.